webclient/timeasy-webclient/src/app/services/project.service.ts

```diff
@@ -29,7 +30,11 @@ export class ProjectService {
     this.resetState();
     this.http.get<Project>(`${this.apiUrl}/${id}`).subscribe({
       next: (project) => {
-        this.state.project.set(project);
+        const transformedProject = {
```

Comment:
hier kannst du dann auch das update von den signals nutzen
```suggestion
       this.state.project.update({...project, deadline: project.deadline ? DateOnly.fromString(project.deadline as unknown as string) : undefined})
```

macht keinen unterschied sparst dir nur die variable